### PR TITLE
Finish up JavaDoc for the 'Json' classes

### PR DESCRIPTION
**Pull request by sbabcoc**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description
This PR finishes up the JavaDoc comments for the classes and methods of Selenium's Json API's. 
These were started in PR #12584 and got merge before I was finished.

### Motivation and Context
The Selenium Json API is a very useful built-in mechanism for consuming and producing JSON representations of Java objects. This facility can be very useful, especially when interacting with GraphQL web service endpoints, but the lack of documentation can result in considerable confusion.
This PR adds JavaDoc headers to classes, methods, and defined constants to assist folks in using the Selenium Json API in their own code.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: java/src/org/openqa/selenium/json/JsonOutput.java [hunk 1]
```diff
@@ -110,159 +112,169 @@ public class JsonOutput implements Closeable {
   private String indentBy = "  ";
   private boolean writeClassName = true;
 
+  /**
```
Comment (shs96c) [inline java/src/org/openqa/selenium/json/JsonOutput.java]: Unless it's really helpful to people modifying the code, we only need to document `public` and `protected` methods.

File: java/src/org/openqa/selenium/json/JsonOutput.java [hunk 2]
```diff
@@ -110,159 +112,169 @@ public class JsonOutput implements Closeable {
...
+        try {
+          appendable.append(str);
+        } catch (IOException e) {
+          throw new JsonException("Unable to write to underlying appendable", e);
+        }
+      };
 
     this.stack = new ArrayDeque<>();
-    this.stack.addFirst(new Empty());
+    this.stack.addFirst(new Root());
```
Comment (shs96c) [inline java/src/org/openqa/selenium/json/JsonOutput.java:129]: This looks suspicious.... Were things failing? Or is this part of the state-machine work?

File: java/src/org/openqa/selenium/json/JsonOutput.java [hunk 3]
```diff
@@ -110,159 +112,169 @@ public class JsonOutput implements Closeable {
...
+      });
 
     this.converters = Collections.unmodifiableMap(builder);
   }
 
   /**
-   * @param enablePrettyPrinting
-   * @return
+   * Specify whether the serialized JSON object should br formatted with line breaks and indention ("pretty printed").
+   * @param enablePrettyPrinting {@code false} for compact format; {@code true} for "pretty printing"
```
Comment (shs96c) [inline java/src/org/openqa/selenium/json/JsonOutput.java]: The formatting here looks really odd. How come the `default` is so far indented?

File: java/src/org/openqa/selenium/json/JsonOutput.java [hunk 4]
```diff
@@ -271,16 +283,22 @@ public JsonOutput setPrettyPrint(boolean enablePrettyPrinting) {
   }
 
   /**
-   * @param writeClassName
-   * @return
+   * Specify whether the serialized JSON object should include a "class" property whose value is the fully-qualified
+   * class name of the Java object being serialized.
+   *
+   * @param writeClassName {@code false} to omit the "class" property; {@code true} to include this property
```
Comment (shs96c) [inline java/src/org/openqa/selenium/json/JsonOutput.java]: For boolean cases, we typically follow the Google convention of documenting as "whether something or rather". In this case "Whether to include the "class" property"

File: java/src/org/openqa/selenium/json/JsonOutput.java [hunk 5]
```diff
@@ -350,35 +383,44 @@ public JsonOutput endArray() {
...
   public JsonOutput write(Object value, int maxDepth) {
     return write0(value, maxDepth, maxDepth);
   }
 
   /**
-   * @param input
-   * @param maxDepth
-   * @param depthRemaining
-   * @return
+   * Serialize the specified Java object as a JSON value.
```
Comment (shs96c) [inline java/src/org/openqa/selenium/json/JsonOutput.java]: We can omit comments for private methods. The question to ask is "does this comment add extra value to people attempting to maintain the code?" If you think the answer is "yes", then please do leave this javadoc in.

File: java/src/org/openqa/selenium/json/JsonOutput.java (hunk 2 above)
Comment (sbabcoc) [reply java/src/org/openqa/selenium/json/JsonOutput.java:129]: No, this just seems like a more descriptive name for the role this class plays. I can revert this change if you prefer.

File: java/src/org/openqa/selenium/json/JsonOutput.java (hunk 3 above)
Comment (sbabcoc) [reply java/src/org/openqa/selenium/json/JsonOutput.java]: This formatting was applied by IntelliJ when I inserted a line break. The `format` script has cleaned everything up now.

File: java/src/org/openqa/selenium/json/JsonOutput.java (hunk 5 above)
Comment (sbabcoc) [reply java/src/org/openqa/selenium/json/JsonOutput.java]: This particular private method probably doesn't benefit much from the JavaDoc header.

File: java/src/org/openqa/selenium/json/JsonOutput.java (hunk 1 above)
Comment (sbabcoc) [reply java/src/org/openqa/selenium/json/JsonOutput.java]: I've removed the JavaDoc headers from all private and package-private methods that don't seem to need them.

File: java/src/org/openqa/selenium/json/JsonOutput.java (hunk 4 above)
Comment (sbabcoc) [reply java/src/org/openqa/selenium/json/JsonOutput.java]: I assume your feedback here refer to the comment for the [writeClassName] parameter. Brevity is the soul of wit. DONE!

File: scripts/format.ps1
Comment (sbabcoc) [inline scripts/format.ps1:1]: This PowerShell script has been brought to you by Google Bard. I asked... it delivered!

**Review by diemol — APPROVED** [review]
Thank you, @sbabcoc!

**Pull request merged**